src/message.rs

```diff
@@ -1377,14 +1404,37 @@ pub(crate) fn rfc724_mid_exists(
         .map_err(Into::into)
 }
 
+/// Returns all MsgIds corresponding to Message-ID
+pub(crate) fn get_all_by_rfc724_mid(
+    context: &Context,
+    rfc724_mid: &str,
+) -> Result<Vec<MsgId>, Error> {
+    ensure!(!rfc724_mid.is_empty(), "empty rfc724_mid");
+
+    let msg_ids = context.sql.query_map(
+        "SELECT id FROM msgs WHERE rfc724_mid=?",
+        params![rfc724_mid],
+        |row| row.get::<_, MsgId>("id"),
+        |ids| {
+            let mut ret: Vec<MsgId> = Vec::new();
+            for id in ids {
+                ret.push(id?);
+            }
+            Ok(ret)
+        },
+    )?;
+    Ok(msg_ids)
+}
+
 pub fn update_server_uid(
     context: &Context,
     rfc724_mid: &str,
     server_folder: impl AsRef<str>,
     server_uid: u32,
 ) {
     match context.sql.execute(
-        "UPDATE msgs SET server_folder=?, server_uid=? WHERE rfc724_mid=?;",
+        "UPDATE msgs SET server_folder=?, server_uid=? \
```

Comment:
i think, the safest way for multi-line-sql-statements is without any backslash or whatever, we did some tests at https://github.com/deltachat/deltachat-core-rust/pull/852/files#diff-9b21422479f64668d6b6d969d83fd28cR1194